app_dart/lib/src/model/appengine/task.dart

```diff
@@ -32,6 +32,7 @@ class Task extends Model {
     this.requiredCapabilities,
     this.reservedForAgentId = '',
     this.stageName,
+    this.buildId,
```

Comment:
Since a task can represent one of Devicelab/LUCI/Cirrus task. How do we know which fields need to be filled when constructing a Devicelab task (or LUCI/Cirrus)?

Reply:
`stageName` tells the difference: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/model/appengine/task.dart#L34

Reply:
But we still don't know what fields need to be filled. Please add constructors for each variant.

Reply:
Added.